compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -150,6 +151,19 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
     Builder b(context);
     SmallVector<NamedAttribute> configItems;
 
+    // Make sure the input number of rows/cols is smaller or equal to the max
+    // number of rows/cols from the device.
+    AMDAIEDeviceModel deviceModel =
+        AMDAIE::getDeviceModel(options.AMDAIETargetDevice);
+    uint32_t maxCoreRows = deviceModel.getNumCoreRows();
+    uint32_t maxCoreCols = deviceModel.getNumCoreCols();
+    if (options.AMDAIENumRows <= 0 || options.AMDAIENumRows > maxCoreRows) {
+      llvm::report_fatal_error("option numRows is out of range\n");
```

Comment:
```suggestion
            llvm::report_fatal_error(llvm::Twine("Invalid number of core rows (") +
                               std::to_string(options.AMDAIENumRows) +
                               "), must be in the range [1, " +
                               std::to_string(maxCoreRows) + "] for device " +
                               stringifyEnum(deviceModel.device));
```

Might save you in the future! A lit test of this would be nice if possible.